analyzers/src/SonarAnalyzer.Common/Json/Parsing/LexicalAnalyzer.cs

```diff
@@ -341,6 +341,11 @@ object BuildResult()
                     return Convert.ToDouble(baseValue) * Math.Pow(10, int.Parse(exponent.ToString()));
                 }
             }
+
+            static object ParseNumber(string number) =>
```

Comment:
Might be an overkill to handle more than long for now.
WDYT?

Reply:
Yes, I think long is fine.